redis/src/main/java/org/apache/zeppelin/redis/command/list/BLPOPCommand.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.redis.command.list;
+
+import org.apache.zeppelin.redis.command.RedisCommand;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+
+/**
+ * Remove and get the first element in a list, or block until one is available
+ */
+public class BLPOPCommand extends RedisCommand {
+
+    public BLPOPCommand(JedisPool clientPool) {
+        super(clientPool);
+    }
+
+    @Override
+    protected String execute(Jedis client, String[] tokens) {
+        String[] keys = new String[tokens.length - 2];
```

Comment:
Please extract this logic to method of utility class or method of RedisCommand. This is duplicated from many places. The logics which manipulate keys/parameters are all candidates to extract.